src/test/java/uk/gov/pay/adminusers/expungeandarchive/resource/ExpungeAndArchiveHistoricalDataResourceIT.java

```diff
@@ -0,0 +1,240 @@
+            .options(wireMockConfig().port(APP.getWireMockPort()))
+            .build();
+
+    LedgerStub ledgerStub;
+
+    @BeforeEach
+    void setUp() {
+        databaseHelper.truncateAllData();
+        ledgerStub = new LedgerStub(wireMockExtension);
+    }
+
+    @Test
+    void shouldDeleteHistoricalUsers() {
+        User userWithCreatedDate = insertUser(now.minusYears(5), null);
+        User userWithLastLoggedInDate = insertUser(now.minusYears(10), now.minusYears(4));
+        User userWithRecentLoggedInDateAndShouldNotBeDeleted = insertUser(now.minusYears(2), now.minusYears(1));
+        User userWithRecentCreatedDateAndShouldNotBeDeleted = insertUser(now.minusYears(2), null);
+
+        assertUserExists(userWithCreatedDate.getId(), userWithLastLoggedInDate.getId(),
+                userWithRecentLoggedInDateAndShouldNotBeDeleted.getId(), userWithRecentCreatedDateAndShouldNotBeDeleted.getId());
+
+        givenSetup().when()
+                .contentType(JSON)
+                .accept(JSON)
+                .post("v1/tasks/expunge-and-archive-historical-data")
+                .then()
+                .statusCode(200);
+
+        assertUserNotExists(userWithCreatedDate.getId(), userWithLastLoggedInDate.getId());
+        assertUserExists(userWithRecentLoggedInDateAndShouldNotBeDeleted.getId(), userWithRecentCreatedDateAndShouldNotBeDeleted.getId());
+    }
+
+    @Test
+    void shouldDeleteHistoricalInvitesAndForgottenPasswords() {
+        User user = userDbFixture(databaseHelper).insertUser();
+
+        ForgottenPasswordDbFixture forgottenPasswordDbFixture = insertForgottenPassword(user.getId(), now.minusYears(5));
+        ForgottenPasswordDbFixture forgottenPasswordThatShouldNotBeDeleted = insertForgottenPassword(user.getId(), now);
+        String code = insertInvite(now.minusYears(5));
+        String codeThatShouldNotBeDeleted = insertInvite(now);
+
+        assertForgottenPasswordsExist(forgottenPasswordDbFixture.getId(), forgottenPasswordThatShouldNotBeDeleted.getId());
+        assertInvitesExist(code, codeThatShouldNotBeDeleted);
+
+        givenSetup().when()
+                .contentType(JSON)
+                .accept(JSON)
+                .post("v1/tasks/expunge-and-archive-historical-data")
+                .then()
+                .statusCode(200);
+
+        assertForgottenPasswordsExist(forgottenPasswordThatShouldNotBeDeleted.getId());
+        assertInvitesExist(codeThatShouldNotBeDeleted);
+
+        List<Map<String, Object>> forgottenPasswordById = databaseHelper.findForgottenPasswordById((Integer) forgottenPasswordDbFixture.getId());
+        assertTrue(forgottenPasswordById.isEmpty());
+
+        Optional<Map<String, Object>> inviteByCode = databaseHelper.findInviteByCode(code);
+        assertTrue(inviteByCode.isEmpty());
+    }
+
+    @Test
+    void shouldArchiveHistoricalServicesAndDetachUsers() throws JsonProcessingException {
```

Comment:
Couple of questions here.
- Does detaching a user mean the user will still be in the `users` table? If so the test should assert that user still exists.
- If a user hasn't logged in for more than `EXPUNGE_USER_DATA_AFTER_DAYS` days but is still attached to a live service, does the user get detached/deleted?